src/js/components/SidebarComponent.jsx

```diff
@@ -0,0 +1,106 @@
+var Link = require("react-router").Link;
+var React = require("react/addons");
+
+var AppListLabelsFilterComponent =
+  require("../components/AppListLabelsFilterComponent");
+var AppListStatusFilterComponent =
+  require("../components/AppListStatusFilterComponent");
+
+var SidebarComponent = React.createClass({
+  displayName: "SidebarComponent",
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    groupId: React.PropTypes.string.isRequired,
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      filters: {
+        filterLabels: [],
+        filterStatus: []
+      }
+    };
+  },
+
+  updateFilterLabels: function (filterLabels) {
+    var state = this.state;
+
+    this.setState(Object.assign(state.filters, {
```

Comment:
Sorry i made a typo. 

You _are_ modifying `this.state.filters` by "merging" it with an object (`{ filterLabels: filterLabels}`). The result of the `Object.assign` call as it's written right now, is a `state.filters` object with an added property. That is not guaranteed to be side-effect free, in my books, which is why the react addons exist...